perf(http): reduce data transfer when using HTTP caching

alan-agius4 opened the pull request:
This commit reduces the property size in the http transfer cache to reduce the page payload.

Before
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "body": "....",
    "headers": {},
    "status": 200,
    "statusText": "OK",
    "url": "http://foo.com/assets/media.json",
    "responseType": "json"
  },
}
</script>
```

Now
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "b": "....",
    "h": {},
    "s": 200,
    "st": "OK",
    "u": "http://foo.com/assets/media.json",
    "rt": "json"
  },
}
</script>
```

HyperLife1119 commented:
How about using tuples?

alan-agius4 commented:
> How about using tuples?

I went with a similar approach that is used for other things like hydration. Also with tuples, it feel like it would be harder to visually read what is written in the cache.

AndrewKushnir commented on packages/common/http/src/transfer_cache.ts:
```diff
@@ -36,12 +36,18 @@ export type HttpTransferCacheOptions = {
-  statusText?: string;
-  url?: string;
-  responseType?: HttpRequest<unknown>['responseType'];
+  /** body */
+  b: any;
```
I think it'd be great to have those keys in consts, something similar to this:

https://github.com/angular/angular/blob/main/packages/core/src/hydration/interfaces.ts#L27-L38

alan-agius4 committed:
perf(http): reduce data transfer when using HTTP caching

This commit reduces the property size in the http transfer cache to reduce the page payload.

Before
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "body": "....",
    "headers": {},
    "status": 200,
    "statusText": "OK",
    "url": "http://foo.com/assets/media.json",
    "responseType": "json"
  },
}
</script>
```

Now
```html
<script id="ng-state" type="application/json">
{
  "4155228514": {
    "b": "....",
    "h": {},
    "s": 200,
    "st": "OK",
    "u": "http://foo.com/assets/media.json",
    "rt": "json"
  },
}
</script>
```

AndrewKushnir reviewed (approved):
Looks great! 👍

alan-agius4 committed:
fixup! perf(http): reduce data transfer when using HTTP caching

dylhunn commented:
This PR was merged into the repository by commit c5e30f1d79f3ebd85cb8061452a897313208bbca.
